Do not unnecessarily connect to the privacyIDEA server

gber opened the pull request.

sunweaver commented on personal/privacyidea/class_PrivacyIdeaUtils.inc:
```diff
@@ -136,6 +134,10 @@ class PrivacyIdeaUtils implements PILog
     {
+        if (!$this->authenticate()) {
+            return;
+        }
```
What will happen if we return from here. Is there already some auth-failure handling (REST API auth against privacyIDEA failing, I mean).

@dzatoah cc

gber replied:
This method is public and there is a TODO to modify this so it fetches tokens directly from PrivacyIDEA without passing them in. However it is currently only used internally and needs to be passed tokens in, so the check can be left out for now until that is actually implemented.

sunweaver commented on personal/privacyidea/class_PrivacyIdeaUtils.inc:
```diff
@@ -176,6 +178,10 @@ class PrivacyIdeaUtils implements PILog
+        if (!$this->authenticate()) {
+            return array();
+        }
+
```
dito here

gber replied:
This is already an existing  failure mode. However while reviewing this again I found a bug in `personal/privacyidea/tokens/class_MFAPaperToken.inc` where an error wasn't handled properly. I've fixed that in another commit.

sunweaver commented on personal/privacyidea/class_PrivacyIdeaUtils.inc:
```diff
@@ -224,6 +230,10 @@ class PrivacyIdeaUtils implements PILog
+        if (!$this->authenticate()) {
+            return array();
+        }
+
```
dito here

gber replied:
The above is equal to the existing failure mode of `piSendRequest` and handled properly.

sunweaver reviewed (commented):
Some feedback needed...

gber commented:
While addressing the review I found another bug in `personal/privacyidea/class_mfaAccount.inc` where failure in `getTokensOfUser` wasn't handled properly, a fix is also included here ([61236a1](https://github.com/gosa-project/gosa-plugins-privacyidea/pull/10/commits/61236a1afe6569b4bb7a6e8a4400352e4b2c14e3)).
